tests/pkg-layering/main.yml

```diff
@@ -307,6 +307,93 @@
       vars:
         package: "{{ g_epel_pkg }}"
 
+- name: Package Layering - GPG Test
+  hosts: all
+  become: true
+
+  tags:
+    - gpg
+
+  tasks:
+    - name: Make /srv/localrepo directory
+      file:
+        path: /srv/localrepo
+        state: directory
+        mode: 0755
+
+    - name: Copy aht-dummy rpm to /srv/localrepo
+      copy:
+         src: '{{ playbook_dir}}/../../rpm/aht-dummy-1.0-1.noarch.rpm'
+         dest: /srv/localrepo
+         owner: root
+         group: root
+         mode: 0644
+
+    - name: Create /etc/yum.repos.d/local.repo
+      file:
+        path: /etc/yum.repos.d/local.repo
+        group: root
+        owner: root
+        mode: 0644
+        state: touch
+
+    - name: Add file contents to local.repo with gpgcheck enabled
+      blockinfile:
+         dest: /etc/yum.repos.d/local.repo
+         marker: ''
+         block: |
+           [local]
+           baseurl=http://localhost
+           gpgcheck=1
+           enabled=1
+
+    - name: Create repo and start http server in container
+      command: >
+         docker run -d -p 80:80
+         -v /srv/localrepo:/srv/localrepo:z
+         -w /srv/localrepo
+         docker.io/miabbott/aht-tools
+         /bin/bash -c 'createrepo_c .; python -m SimpleHTTPServer 80'
+
+    - name: Wait for port 80 to open
+      wait_for:
+        port: 80
+        delay: 5
+        timeout: 30
+
+    - name: Attempt to install unsigned rpm with gpgcheck enabled
+      command: rpm-ostree install aht-dummy
+      register: ros_install
+      failed_when: ros_install|success
```

Comment:
Should there be an additional check here to see if the error message returned is about GPG checking?

Reply:
I can add that.